source/src/microapp/cs_MicroappController.cpp

```diff
@@ -192,6 +193,14 @@ MicroappController::MicroappController() : EventListener(), _callCounter(0), _mi
 		_bleIsr[i].id = 0;
 	}
 
+	_watchdogDataLength = sizeof(watchdog_data_t);
+
+	_watchdogData                               = new uint8_t[_watchdogDataLength];
+	((watchdog_data_t*)_watchdogData)->flags    = (uint8_t)MicroappWatchdogFlags::CS_WATCHDOG_MICROAPP_BUILD;
+	for (int i = 0; i < MAX_MICROAPPS; ++i) {
+		((watchdog_data_t*)_watchdogData)->appRunning[i] = (uint8_t)MicroappWatchdogState::CS_WATCHDOG_MICROAPP_NOT_RUNNING;
```

Comment:
Why have this on the heap?
And are all the casts needed?

Reply:
Why not on the heap? Is there a policy against it?